src/pkcs15init/pkcs15-myeid.c

```diff
@@ -510,10 +510,10 @@ myeid_store_key(struct sc_profile *profile, struct sc_pkcs15_card *p15card,
 				keybits = key_info->field_length;
 			else
 				key_info->field_length = keybits;
-			break;
-			
+
 			if (sc_card_find_ec_alg(p15card->card, keybits, &prkey->u.ec.params.id) == NULL)
 				LOG_TEST_RET(ctx, SC_ERROR_INVALID_ARGUMENTS, "Unsupported algorithm or key size");			
+			break;
```

Comment:
@mtrojnar have a look on the full diff of the file. Originally, the `break` was before the `sc_card_find_ec_alg()` so it was not even called.

Also if `sc_card_find_ec_alg()` succeed, it would fall through to the `default` of the switch, which is certainly not intended.